### PR TITLE
feat: Upgrade dependencies to Keptn 0.16

### PR DESCRIPTION
**Pull request by christian-kreuzberger-dtx**
* Introduce compatibility with Keptn 0.16
* Update go-utils to current master branch (includes this PR https://github.com/keptn/go-utils/pull/480 in order to overcome a problem with Keptn 0.16 and job-executor-service)
* Updated integration test matrix
* Added gitea and a gitea-upstream-provisioner (as git upstream is mandatory for Keptn 0.16) to integration tests
* Updated installation action to use helm-chart, in order to configure the upstream provisioner

Integration Test Run: https://github.com/keptn-contrib/job-executor-service/actions/runs/2582486075

### CODE REVIEW

**Comment by christian-kreuzberger-dtx** [comment]
# Integration Test Report

| Test                                                         | 0.13.5-netpolicy-false-jobnetpolicy-false | 0.13.5-netpolicy-false-jobnetpolicy-true | 0.13.5-netpolicy-true-jobnetpolicy-false | 0.13.5-netpolicy-true-jobnetpolicy-true | 0.14.2-netpolicy-false-jobnetpolicy-false | 0.14.2-netpolicy-false-jobnetpolicy-true | 0.14.2-netpolicy-true-jobnetpolicy-false | 0.14.2-netpolicy-true-jobnetpolicy-true | 0.15.1-netpolicy-false-jobnetpolicy-false | 0.15.1-netpolicy-false-jobnetpolicy-true | 0.15.1-netpolicy-true-jobnetpolicy-false | 0.15.1-netpolicy-true-jobnetpolicy-true | 0.16.0-netpolicy-false-jobnetpolicy-false | 0.16.0-netpolicy-false-jobnetpolicy-true | 0.16.0-netpolicy-true-jobnetpolicy-false | 0.16.0-netpolicy-true-jobnetpolicy-true |
| :----------------------------------------------------------- | :---------------------------------------: | :--------------------------------------: | :--------------------------------------: | :-------------------------------------: | :---------------------------------------: | :--------------------------------------: | :--------------------------------------: | :-------------------------------------: | :---------------------------------------: | :--------------------------------------: | :--------------------------------------: | :-------------------------------------: | :---------------------------------------: | :--------------------------------------: | :--------------------------------------: | :-------------------------------------: |
| TestEnvironmentVariables                                     |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestResourceFiles                                            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestHelloWorldDeployment                                     |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestJobCleanupWithSmallTTL                                   |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestJobCleanupWith0TTLMultipleJobs                           |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestJobNetworkPolicy_NoIngress                               |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestJobNetworkPolicy_NoEgress                                |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestLocust                                                   |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestJobExecutorServiceNoIngress                              |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestJobDeadline/No_deadline_set_-_job_completes_successfully |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestJobDeadline/Deadline_too_short_-_job_fails               |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestJobDeadline/Deadline_long_enough_-_job_succeeds          |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |
| TestJobDeadline                                              |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |            :heavy_check_mark:             |            :heavy_check_mark:            |            :heavy_check_mark:            |           :heavy_check_mark:            |

File: skaffold.yaml [hunk 1]
```diff
@@ -54,8 +54,6 @@ deploy:
               k8sMasterPort: 0
         overrides:
           distributor:
-            image:
```
Comment (TannerGabriel) [inline skaffold.yaml:57]: Any reason why we are removing the version here?

Comment (christian-kreuzberger-dtx) [reply skaffold.yaml:57]: it's already defined in the chart/values.yaml, this information is therefore redundant.

**Review by TannerGabriel — APPROVED** [review]
LGTM

File: .github/workflows/integration-tests.yaml [hunk 2]
```diff
@@ -131,6 +181,9 @@ jobs:
           kubectl get configmaps,deployments,pods,networkpolicy,serviceaccounts,role,rolebindings,events -n ${JES_NAMESPACE} -o json > k8s_debug/k8s_jes_objects.json
           kubectl logs -n keptn -l app.kubernetes.io/instance=keptn --prefix=true --previous=false --all-containers > k8s_debug/k8s_keptn_logs.txt || true
           kubectl logs -n ${JES_NAMESPACE} deployment/job-executor-service --prefix=true --previous=false --all-containers > k8s_debug/k8s_jes_logs.txt || true
+          kubectl logs deployment/keptn-gitea-provisioner-service --prefix=true --previous=false --all-containers > k8s_debug/k8s_gitea_provisioner_logs.txt
```
Comment (Raffy23) [inline .github/workflows/integration-tests.yaml:184]: We should probably use the same trick as the statements above to ensure that the pipeline doesn't fail completely if something happened to the deployment
```suggestion
          kubectl logs deployment/keptn-gitea-provisioner-service --prefix=true --previous=false --all-containers > k8s_debug/k8s_gitea_provisioner_logs.txt || true
```

File: .github/workflows/integration-tests.yaml [hunk 3]
```diff
@@ -131,6 +181,9 @@ jobs:
...
+          kubectl get statefulsets,configmaps,pods,networkpolicy,serviceaccounts,role,rolebindings,events,services -n ${GITEA_NAMESPACE} -o json > k8s_debug/k8s_objects_gitea.json
+          kubectl logs statefulsets/gitea --prefix=true --previous=false --all-containers -n ${GITEA_NAMESPACE} > k8s_debug/k8s_logs_gitea.txt
```
Comment (Raffy23) [inline .github/workflows/integration-tests.yaml:186]: Same here
```suggestion
          kubectl logs statefulsets/gitea --prefix=true --previous=false --all-containers -n ${GITEA_NAMESPACE} > k8s_debug/k8s_logs_gitea.txt || true
```

File: .github/workflows/integration-tests.yaml (hunk 2 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/integration-tests.yaml]: Done

File: .github/workflows/integration-tests.yaml (hunk 3 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/integration-tests.yaml]: Done

**Review by Raffy23 — APPROVED** [review]
LGTM

**Pull request merged**